lib/config.py

```diff
@@ -77,62 +79,88 @@ class ConfigParser(object):
     Precedence is:
     cmdline > config file > argparse defaults
     """
-    def __init__(self):
-        self._CONF = self._parse_arguments()
+    def __init__(self, current_working_dir=None):
+        self.parser = argparse.ArgumentParser()
+        self._CONF = None
+        if current_working_dir:
+            self._cwd = current_working_dir
+        else:
+            self._cwd = os.getcwd()
+
+        self._setup_config_parser_options()
 
     @property
     def CONF(self):
         return self._CONF
 
-    def _parse_config(self, config_file):
+    def _setup_config_parser_options(self):
+        """
+	Configures the argument parser object to match the expected
+            configuration.
+        """
+        self.parser.add_argument('--config-file', '-c',
+                                 help='Path of the configuration file for build '
+                                     'scripts',
+                                 # NOTE(maurosr): move this to /etc in the future
+                                 default='./config.yaml')
+        self.parser.add_argument('--packages', '-p',
+                                 help='Packages to be built',
+                                 nargs='*')
+        self.parser.add_argument('--log-file', '-l',
+                                 help='Log file',
+                                 default='/var/log/host-os/builds.log')
+        self.parser.add_argument('--verbose', '-v',
+                                 help='Set the scripts to be verbose',
+                                 action='store_true')
+        self.parser.add_argument('--result-dir', '-r',
+                                 help='Directory to save the RPMs.',
+                                 default=os.path.join(self._cwd, 'result'))
+        self.parser.add_argument('--repositories-path', '-R',
+                                 help='Directory where to clone code repositories',
+                                 default='/var/lib/host-os/repositories')
+        self.parser.add_argument('--keep-builddir',
+                                 help='Keep build directory and its logs and '
+                                 'artifacts.', action='store_true')
+        self.parser.add_argument('--build-versions-repository-url',
+                                 help='Build versions repository URL')
+        self.parser.add_argument('--build-version',
+                                 help='Select build version from versions '
+                                 'repository')
+
+    def parse_arguments_list(self, args):
+        """
+        Parses the arguments provided in the argument list and returns
+            the result object.
+        """
+        result = self.parser.parse_args(args)
+        return vars(result)
+
+    def parse_config_file(self, config_file_path):
+        """
+	Parse the configuration file and returns a dictionary containing the
```

Comment:
@snowcrshd I suggest you make a pass through pycodestyle just to make sure no more issues like this go unnoticed.